presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -286,72 +293,70 @@ public void flushCache()
+        Map<String, HiveColumnStatistics> columnStatistics = delegate.getTableColumnStatistics(hiveTableName.getDatabaseName(), hiveTableName.getTableName(), columnNames);
 
-        ImmutableMap.Builder<TableColumnStatisticsCacheKey, Optional<HiveColumnStatistics>> resultMap = ImmutableMap.builder();
+        ImmutableMap.Builder<TableColumnStatisticsCacheKey, HiveColumnStatistics> resultMap = ImmutableMap.builder();
         for (TableColumnStatisticsCacheKey key : keys) {
-            if (!columnStatistics.isPresent() || !columnStatistics.get().containsKey(key.getColumnName())) {
-                resultMap.put(key, Optional.empty());
-            }
-            else {
-                resultMap.put(key, Optional.of(columnStatistics.get().get(key.getColumnName())));
-            }
+            HiveColumnStatistics statistics = columnStatistics.get(key.getColumnName());
+            resultMap.put(key, statistics == null ? EMPTY_COLUMN_STATISTICS : statistics);
         }
         return resultMap.build();
     }
 
     @Override
-    public Optional<Map<String, Map<String, HiveColumnStatistics>>> getPartitionColumnStatistics(String databaseName, String tableName, Set<String> partitionNames, Set<String> columnNames)
+    public Map<String, Map<String, HiveColumnStatistics>> getPartitionColumnStatistics(String databaseName, String tableName, Set<String> partitionNames, Set<String> columnNames)
     {
         List<PartitionColumnStatisticsCacheKey> cacheKeys = partitionNames.stream()
                 .flatMap(
                         partitionName -> columnNames.stream().map(
                                 columnName -> new PartitionColumnStatisticsCacheKey(databaseName, tableName, partitionName, columnName)))
-                .collect(toList());
-        Map<PartitionColumnStatisticsCacheKey, Optional<HiveColumnStatistics>> cacheValues = getAll(partitionColumnStatisticsCache, cacheKeys);
+                .collect(toImmutableList());
+        Map<PartitionColumnStatisticsCacheKey, HiveColumnStatistics> cacheValues = getAll(partitionColumnStatisticsCache, cacheKeys);
 
         ImmutableMap.Builder<String, Map<String, HiveColumnStatistics>> partitionsMap = ImmutableMap.builder();
         for (String partitionName : partitionNames) {
             ImmutableMap.Builder<String, HiveColumnStatistics> columnsMap = ImmutableMap.builder();
             for (String columnName : columnNames) {
-                Optional<HiveColumnStatistics> cacheValue = cacheValues.get(new PartitionColumnStatisticsCacheKey(databaseName, tableName, partitionName, columnName));
-                if (cacheValue.isPresent()) {
-                    columnsMap.put(columnName, cacheValue.get());
+                HiveColumnStatistics hiveColumnStatistics = cacheValues.get(new PartitionColumnStatisticsCacheKey(databaseName, tableName, partitionName, columnName));
+                if (hiveColumnStatistics != null && hiveColumnStatistics != EMPTY_COLUMN_STATISTICS) {
```

Comment:
- when `hiveColumnStatistics` can be null?
- as above, HiveColumnStatistics cannot, normally, be compared by `==`, it's more a value object. Please avoid that.